src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateInstanceEventHandler.java

```diff
@@ -148,6 +158,40 @@ public CompletableFuture<DataImportEventPayload> handle(DataImportEventPayload d
     return future;
   }
 
+  private Future<Instance> fill001And999IFields(Instance instance, Record record) {
+    AdditionalFieldsUtil.fill001FieldInMarcRecord(record, instance.getHrid());
+    return AdditionalFieldsUtil.addFieldToMarcRecord(record, TAG_999, 'i', instance.getId())
+      ? Future.succeededFuture(instance)
+      : Future.failedFuture(format("Failed to add instance id '%s' to record with id '%s'", instance.getId(), record.getId()));
+  }
+
+  private Future<Instance> saveRecordInSrsAndHandleResponse(DataImportEventPayload payload, Record record,
+                                                            Instance instance, InstanceCollection instanceCollection) {
+    Promise<Instance> promise = Promise.promise();
+    getSourceStorageRecordsClient(payload).postSourceStorageRecords(record)
+      .onComplete(ar -> {
+        var result = ar.result();
+        if (ar.succeeded() && result.statusCode() == HttpStatus.HTTP_CREATED.toInt()) {
+          payload.getContext().put(EntityType.MARC_BIBLIOGRAPHIC.value(), result.bodyAsJsonObject().encode());
+          LOGGER.info("Created MARC record in SRS with id: '{}', instanceId: '{}', from tenant: {}",
```

Comment:
Let's add here jobExecutionId to the log. Smth like: "Created MARC record in SRS with id: '{}', instanceId: '{}', from tenant: {} for jobExecutionId: {}"